docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala

```diff
@@ -71,6 +71,50 @@ class HttpClientExampleSpec extends WordSpec with Matchers with CompileOnlySpec
     //#manual-entity-consume-example-2
   }
 
+  "manual-entity-consume-example-3" in compileOnlySpec {
+    //#manual-entity-consume-example-3
+    import scala.concurrent.duration._
+    import scala.concurrent.Future
+
+    import akka.NotUsed
+    import akka.actor.ActorSystem
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model._
+    import akka.stream.ActorMaterializer
+    import akka.util.ByteString
+    import akka.stream.scaladsl.{Flow, Sink, Source}
+
+    implicit val system = ActorSystem()
+    implicit val dispatcher = system.dispatcher
+    implicit val materializer = ActorMaterializer()
+
+    case class ExamplePerson(name: String)
+    def parse(line: ByteString): ExamplePerson = ???
+
+    val requests: Source[HttpRequest, NotUsed] = ???
+
+    val processorFlow: Flow[ExamplePerson, Long, NotUsed] = ???
+
+    // Run and completely consume a single akka http request
+    def runRequest(req: HttpRequest): Future[ExamplePerson] =
+      Http()
+        .singleRequest(req)
+        .flatMap(_.entity.toStrict(3.seconds))
+        .flatMap { e =>
+          e.dataBytes
+            .runFold(ByteString.empty) { case (acc, b) => acc ++ b }
+            .map(parse)
+        }
+
+    // Run each akka http flow to completion, then continue processing
+    requests
+      .mapAsync(2)(runRequest)
```

Comment:
why 2?

Reply:
What do you think about having the same value for the Java example instead of 1?

Reply:
Done =p

Reply:
Perhaps mention in a comment that this is a memory vs parallelism trade-off, since the more parallelism you allow, the more `ExamplePerson`s you will hold in memory?